daily_milestone.go

```diff
@@ -1,29 +1,225 @@
 }
 
-//func get_milestones(base_url string, token string, project_id string)[]string{}
-//func create_milestone_data(advance int) []string{}
-//func create_milestones(base_url string, token string, project_id string, milestones string) string{}
+// CreateMilestoneData is used to check the due date using the time package of python
+func createMilestoneData(advance int) []string {
+	today := time.Now().Local()
+	date := today.AddDate(0, 0, advance)
+	list := []string{}
+	/*for i := 0; i < advance; i++ {
+		date := today.AddDate(0, 0, i)                          // To Format to ISOFormat and it converts to string so can be used in list directly
+		datelist := []string{"Title: ", date, "due_date", date} // Was unable to get a map in a list so made this
+		y := strings.Join(datelist, ",")
+		list = append(list, y)
+	}*/
+	fmt.Println(date)
+	fmt.Println(list)
+	return list
+}
+
+func createMilestones(baseURL string, token string, projectID string, milestones []string) string {
+	strurl := []string{"https://", baseURL, "/projects/", projectID, "/milestones"}
+	url := strings.Join(strurl, "")
+	client := &http.Client{}
+	for _, m := range milestones {
+		mbyte := bytes.NewReader([]byte(m))
+		req, err := http.NewRequest("POST", url, mbyte)
+		if err != nil {
+			logger.Println(err)
+			break
+		}
+		req.Header.Add("PRIVATE-TOKEN", token)
+		resp, err := client.Do(req)
+		if err != nil {
+			logger.Println(err)
+			break
+		}
+		defer resp.Body.Close()
+	}
+	return ("Milestones Created: " + strings.Join(milestones, ""))
+}
 
 func main() {
-	var apiKey, apiBase, namespace, project string
-	var advance int
-	flag.StringVar(&apiKey, "token", " ", "Gitlab api key/token.")
-	flag.StringVar(&apiBase, "base_url", " ", "Gitlab api base url")
-	flag.StringVar(&namespace, "namespace", " ", "Namespace to use in Gitlab")
-	flag.StringVar(&project, "projectname", " ", "Project to use in Gitlab")
-	flag.IntVar(&advance, "Advance", 30, "Define timeframe to generate milestones in advance.")
-	flag.Parse()
+	// Declaring variables for flags
+	var Token, APIBase, Namespace, Project string
+	var Advance int
+	APIBase = ""
+	// Command Line Parsing Starts
+	flag.StringVar(&Token, "Token", "bVYFTJaYtgAZesSofKbq", "Gitlab api key/token.")
+	flag.StringVar(&APIBase, "baseURL", "dev.seetheprogress.eu", "Gitlab api base url")
+	flag.StringVar(&Namespace, "Namespace", "okkur", "Namespace to use in Gitlab")
+	flag.StringVar(&Project, "ProjectName", "dailymile_test", "Project to use in Gitlab")
+	flag.IntVar(&Advance, "Advance", 30, "Define timeframe to generate milestones in advance.")
+	flag.Parse() //Command Line Parsing Ends
+
+	// Initializing logger
+	LoggerSetup(os.Stdout)
+	// Calling getProjectID
+	projectID, err := getProjectID(APIBase, Token, Project, Namespace)
+	if err != nil {
+		logger.Println(err)
+	}
+	newMilestone, err := getMilestones(APIBase, Token, projectID)
+	if err != nil {
+		logger.Println(err)
+	}
+	oldMilestone := createMilestoneData(Advance)
```

Comment:
createMilestoneData feels like it would create the newMilestones ;)

Reply:
```python
def create_milestones(base_url, token, project_id, milestones):
    headers = {"PRIVATE-TOKEN": token}
    url = 'https://' + base_url + \
        '/projects/' + str(project_id) + '/milestones'
    for m in milestones:
        requests.post(url, data=m, headers=headers)

    return "Milestones created: " + str(milestones)
```

Reply:
Can't understand this function :(

Reply:
```python

def create_milestone_data(advance):
    today = datetime.date.today()
    list = []
    for i in range(advance):
        date = today + datetime.timedelta(days=+i)
        date_iso = date.isoformat()
        list.append({'title': date_iso, 'due_date': date_iso})

    return list

```

Also can you tell me about using date_iso with title as well? shouldn't it be title shouldn't it have another input argument?

Reply:
Basically we are creating one milestone with the title and the due date set to the iso_date such as 2017-01-01.

So the create_milestones function in python is as far as I see correctly done with your createMilestones func.

My issue was mostly with the naming of oldMilestones vs newMilestones. I would assume we create newMilestones with the advance and use oldMilestones (aka the one already on the server).